sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueAysncAPITests.groovy

```diff
@@ -0,0 +1,548 @@
+            QueueTestHelper.assertExceptionStatusCodeAndMessage(it, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+        }
+    }
+
+    def "Delete message from queue async client"() {
+        given:
+        queueAsyncClient.create().block()
+        queueAsyncClient.enqueueMessage("test message 1").block()
+        queueAsyncClient.enqueueMessage("test message 2").block()
+        queueAsyncClient.enqueueMessage("test message 3").block()
+        def dequeueMsg = queueAsyncClient.dequeueMessages().blockFirst()
+        when:
+        def getPropertiesVerifier = StepVerifier.create(queueAsyncClient.getProperties())
+        def deleteMsgVerifier = StepVerifier.create(queueAsyncClient.deleteMessage(dequeueMsg.messageId(), dequeueMsg.popReceipt()))
+        def getPropertiesAfterVerifier = StepVerifier.create(queueAsyncClient.getProperties())
+        then:
+        getPropertiesVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 200)
+            it.value().approximateMessagesCount() == 3
+        }.verifyComplete()
+        deleteMsgVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 204)
+        }.verifyComplete()
+        getPropertiesAfterVerifier.assertNext {
+            QueueTestHelper.assertResponseStatusCode(it, 200)
+            it.value().approximateMessagesCount() == 2
+        }.verifyComplete()
+    }
+
+    @Unroll
+    def "Delete message invalid args from queue async client"() {
+        given:
+        queueAsyncClient.create().block()
+        def expectMsg = "test message"
+        queueAsyncClient.enqueueMessage(expectMsg).block()
+        def dequeueMessage = queueAsyncClient.dequeueMessages().blockFirst()
+        when:
+        def deleteMessageId = messageId ? dequeueMessage.messageId() : dequeueMessage.messageId() + "Random"
+        def deletePopReceipt = popReceipt ? dequeueMessage.popReceipt() : dequeueMessage.popReceipt() + "Random"
+        def deleteMsgVerifier = StepVerifier.create(queueAsyncClient.deleteMessage(deleteMessageId, deletePopReceipt))
+        then:
+        deleteMsgVerifier.verifyErrorSatisfies {
+            QueueTestHelper.assertExceptionStatusCodeAndMessage(it, statusCode, errMsg)
+        }
+        where:
+        messageId | popReceipt | statusCode | errMsg
+        true      | false      | 400        | StorageErrorCode.INVALID_QUERY_PARAMETER_VALUE
+        false     | true       | 404        | StorageErrorCode.MESSAGE_NOT_FOUND
+        false     | false      | 400        | StorageErrorCode.INVALID_QUERY_PARAMETER_VALUE
+    }
+
+    def "Delete message error from async client"() {
+        given:
+        queueAsyncClient.create().block()
+        queueAsyncClient.enqueueMessage("test update message").block()
+        def dequeueMsg = queueAsyncClient.dequeueMessages().blockFirst()
+        queueAsyncClient.delete().block()
+        QueueTestHelper.sleepInRecordMode(Duration.ofSeconds(30))
```

Comment:
Write up a follow-up issue to add a `sleepInRecordMode` somewhere centralized, I feel there may be a lot of places where we could leverage this to speed up our playback tests.

Reply:
I have it before. 
Based on the conversation with Rick, we both think the sdk tests here is basically testing the API to REST call layer. We do not test any service behavior. Like, service will handle it right when create immediately after deletion. In the other word, if this kind of behavior is not correct, service needs to handle it in their tests, there is nothing much sdk can do.

The principal applied to getClient. We do not getProperties over there as we do not make any service call, so no need to do end-end testing.